pkg/controller/resourcegraphdefinition/controller_status.go

```diff
@@ -40,46 +40,39 @@ func NewStatusProcessor() *StatusProcessor {
 		state:      v1alpha1.ResourceGraphDefinitionStateActive,
 	}
 }
-
-// setDefaultConditions sets the default conditions for an active resource graph definition
 func (sp *StatusProcessor) setDefaultConditions() {
 	sp.conditions = []v1alpha1.Condition{
-		newReconcilerReadyCondition(metav1.ConditionTrue, ""),
-		newGraphVerifiedCondition(metav1.ConditionTrue, ""),
-		newCustomResourceDefinitionSyncedCondition(metav1.ConditionTrue, ""),
+		newReconcilerReadyCondition(metav1.ConditionTrue, "ReconcilerAvailable"),
+		newGraphVerifiedCondition(metav1.ConditionTrue, "GraphValidated"),
+		newCustomResourceDefinitionSyncedCondition(metav1.ConditionTrue, "CRDSynced"),
 	}
 }
-
-// processGraphError handles graph-related errors
 func (sp *StatusProcessor) processGraphError(err error) {
 	sp.conditions = []v1alpha1.Condition{
-		newGraphVerifiedCondition(metav1.ConditionFalse, err.Error()),
-		newReconcilerReadyCondition(metav1.ConditionUnknown, "Faulty Graph"),
-		newCustomResourceDefinitionSyncedCondition(metav1.ConditionUnknown, "Faulty Graph"),
+		newGraphVerifiedCondition(metav1.ConditionFalse, "GraphValidationFailed"),
+		newReconcilerReadyCondition(metav1.ConditionUnknown, "GraphIssue"),
+		newCustomResourceDefinitionSyncedCondition(metav1.ConditionUnknown, "GraphIssue"),
 	}
 	sp.state = v1alpha1.ResourceGraphDefinitionStateInactive
 }
 
-// processCRDError handles CRD-related errors
 func (sp *StatusProcessor) processCRDError(err error) {
 	sp.conditions = []v1alpha1.Condition{
-		newGraphVerifiedCondition(metav1.ConditionTrue, ""),
-		newCustomResourceDefinitionSyncedCondition(metav1.ConditionFalse, err.Error()),
-		newReconcilerReadyCondition(metav1.ConditionUnknown, "CRD not-synced"),
+		newGraphVerifiedCondition(metav1.ConditionTrue, "GraphValidated"),
+		newCustomResourceDefinitionSyncedCondition(metav1.ConditionFalse, "CRDNotSynced"),
+		newReconcilerReadyCondition(metav1.ConditionUnknown, "CRDNotReady"),
 	}
 	sp.state = v1alpha1.ResourceGraphDefinitionStateInactive
 }
 
-// processMicroControllerError handles microcontroller-related errors
 func (sp *StatusProcessor) processMicroControllerError(err error) {
 	sp.conditions = []v1alpha1.Condition{
-		newGraphVerifiedCondition(metav1.ConditionTrue, ""),
-		newCustomResourceDefinitionSyncedCondition(metav1.ConditionTrue, ""),
-		newReconcilerReadyCondition(metav1.ConditionFalse, err.Error()),
+		newGraphVerifiedCondition(metav1.ConditionTrue, "GraphValidated"),
```

Comment:
I am shocked setting reason to "" worked before, this violates the built in type validation of conditions if you use the upstream open api.